src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -777,28 +767,15 @@ private Future<Map<String, List<Piece>>> processItemsUpdate(Map<String, List<Pie
   protected abstract Future<Boolean> receiveInventoryItemAndUpdatePiece(JsonObject item, Piece piece, RequestContext locationContext);
 
   private boolean holdingUpdateOnCheckinReceiveRequired(Piece piece, CompositePoLine poLine) {
+    // In case of affiliation change current holding is irrelevant because we will create a new holding in
+    // another tenant, hence we need to block the creation of instances per affiliation-changed piece here
     boolean isHoldingUpdateRequired;
     if (piece.getFormat() == Piece.Format.ELECTRONIC) {
       isHoldingUpdateRequired = PoLineCommonUtil.isHoldingUpdateRequiredForEresource(poLine);
     } else {
       isHoldingUpdateRequired = PoLineCommonUtil.isHoldingUpdateRequiredForPhysical(poLine);
     }
-    String locationId = getLocationId(piece);
-    String holdingId = getHoldingId(piece);
-    return isHoldingUpdateRequired && (StringUtils.isNotEmpty(locationId) || StringUtils.isNotEmpty(holdingId));
-  }
-
-  private String buildProcessedHoldingKey(Piece piece, String instanceId) {
```

Comment:
This method prevented holding creation if a new holding is requested by the user, so it was removed altogether